pkg/apis/numaplane/v1alpha1/pipelinerollout_types.go

```diff
@@ -181,6 +203,12 @@ func (pipelineRollout *PipelineRollout) ResetUpgradingChildStatus(upgradingPipel
 		},
 	}
 
+	// only set analysisStatus if Analysis is set
+	if len(pipelineRollout.GetAnalysis().Templates) > 0 {
+		upgradingPipelineStatus.Analysis = AnalysisStatus{}
```

Comment:
We should just clear this regardless of whether any there are any Templates configured